main.go

```diff
@@ -130,24 +124,115 @@ func actionRun(c *cli.Context) error {
 		}
 
 		// Save the config file path for future use
-		s.configFilePath = configPath
+		s.SetConfigFilePath(configPath)
 
 		// Configure all modules
-		err = s.loadConfig(conf)
+		err = s.LoadConfig(conf)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Anonymously collect usage metrics if not explicitly disabled
+	if !disableMetrics {
+		go s.RoutineMetrics()
+	}
+
+	s.Routes()
+	return s.Start(port, grpcPort)
+}
+
+func actionStart(c *cli.Context) error {
+	// Load cli flags
+	port := c.String("port")
+	grpcPort := c.String("grpc-port")
+	natsPort := c.Int("nats-port")
+	clusterPort := c.Int("cluster-port")
+	configPath := c.String("config")
+	isProd := c.Bool("prod")
+	disableMetrics := c.Bool("disable-metrics")
+	disableNats := c.Bool("disable-nats")
+	seeds := c.String("seeds")
+
+	// Project and env cannot be changed once space cloud has started
+	s := server.New(isProd)
+
+	if configPath == "none" {
+		configPath = "./config.yaml"
+	}
+	// Load config file
+	conf, err := config.LoadConfigFromFile(configPath)
+	// If config file does not exists then trigger the generate config flow
+	if err != nil {
+		err := config.GenerateConfig(configPath, true)
+		if err != nil {
+			return nil
+		}
+		temp, err := config.LoadConfigFromFile(configPath)
 		if err != nil {
 			return err
 		}
+
+		conf = temp
```

Comment:
We don't need this line

Reply:
This line will be needed as conf is been used below in `s.LoadConfig(conf)`

Reply:
If line 172 is changed, this wouldn't be required

Reply:
Cool got it!

Reply:
Done